pilot/pkg/bootstrap/server.go

```diff
@@ -1011,7 +1011,11 @@ func (s *Server) createPeerCertVerifier(tlsOptions TLSOptions) (*spiffe.PeerCert
 		}
 	} else {
 		if s.RA != nil {
-			rootCertBytes = append(rootCertBytes, s.RA.GetCAKeyCertBundle().GetRootCertPem()...)
+			if strings.HasPrefix(features.PilotCertProvider, constants.CertProviderKubernetesSignerPrefix) {
```

Comment:
if this is "RA" then why aren't we using s.RA here?

Reply:
There are two cases for root certs when s.RA is not nil:
1.  Using root certs in meshconfig 
1.  Mount `external-ca-cert` secret and use it as the RA's root certs  -- previous supported https://istio.io/latest/docs/tasks/security/cert-management/custom-ca-k8s/#part-2-using-custom-ca

Reply:
Not sure if this is the best logic: the bundle here is supposed to represent the mesh cabundle. The "istiodCertBundleWatcher" is only meant to watch the caBundle corresponding to the Istiod cert, which may be different from the RA's cabundle. There are use cases where we could use Istiod as an RA to sign workload certificates, but use self-signed Istiod certificates.

Reply:
Yes, so this is the reason why we add a condition here  `if strings.HasPrefix(features.PilotCertProvider, constants.CertProviderKubernetesSignerPrefix)`, so when istiod is acting as RA and istiod's certificate is signed by k8s signer we will use `istiodCertBundleWatcher` . The `istiodCertBundleWatcher` is set in https://github.com/istio/istio/blob/f26bfc01f2ce4254b0988e48980dbb9809be4397/pilot/pkg/bootstrap/certcontroller.go#L137 in this case to use the one from meshconfig.

Reply:
can we just pick up the certificates from the ra only with the latest set of changes?

Reply:
changed to getRootCertFromRA

Reply:
Don't see the change here yet..were you planning to add it?

Reply:
changed to get from RA directly now.